rpc/blockchain.go

```diff
@@ -41,10 +42,29 @@ func NewPublicBlockchainAPI(hmy *hmy.Harmony, version Version) rpc.API {
 	}
 }
 
+// ChainId returns the chain id of the chain - required by MetaMask
+func (s *PublicBlockchainService) ChainId(ctx context.Context) (interface{}, error) {
+	// Format return base on version
+	switch s.version {
+	case V1:
+		return hexutil.Uint64(s.hmy.ChainID), nil
+	case V2:
+		return s.hmy.ChainID, nil
+	case Eth:
+		shardID, err := nodeconfig.GetDefaultConfig().ShardIDFromConsensusKey()
+		if err != nil {
+			return nil, err
+		}
+		return hexutil.Uint64(nodeconfig.GetDefaultConfig().GetNetworkType().ChainConfig().EthCompatibleChainID.Uint64() + uint64(shardID)), nil
```

Comment:
No need to add shard ID, EthCompatibleChainID is already updated to be the correct chainID (added with shardID) in the beginning of node start.  search for this method `UpdateEthChainIDByShard`